src/fetchers/opensea.js

```diff
@@ -32,39 +32,71 @@ export const fetchCollection = async (chainId, { contract, tokenId }) => {
               },
       });
 
-      data = assetResponse.data;
+      data = assetResponse.data.asset_events[0].asset;
     } catch (error) {
-      logger.error(
+      logger.warn(
         "opensea-fetcher",
-        `fetchCollection retrieve asset error. chainId:${chainId}, contract:${contract}, tokenId:${tokenId}, message:${
+        `fetchCollection retrieve event error. chainId:${chainId}, contract:${contract}, tokenId:${tokenId}, message:${
           error.message
         },  status:${error.response?.status}, data:${JSON.stringify(error.response?.data)}`
-      );
+      )
+    }
 
-      // Try to get the collection only based on the contract.
-      if (error.response?.status === 404) {
+    if (!data) {
+      // Fall back to the asset endpoint if events response fails
```

Comment:
i think the fallback should only be for 4, 5, 1 (coz these apis will fail for the other networks) or just do the same check as you did above [4,5], otherwise it will will go to testnet for polygon, optimisim